Feature/transfer screen

n13 opened the pull request:
### 🗃 Github Issue Or Explanation for this PR. (What is it supposed to do and Why is needed)

Transfer screen looked bad because keyboard was overlapping the "send" button

Solution: Make transfer screen less tall by moving available balance under the entry field

Small change to list of items overview in merchant mode

Fix: Transition to receive mode - was horribly jarring (black/white) on the phone - this was invisible on the simulator. 
Cause: sizeBox(0) returned when items are loading
Fix: Show progress indicator, make the background white for the SafeArea.

### ✅ Checklist

- [x] I have tested all my changes.

### 🕵️‍♂️ Notes for Code Reviewer

This change affects all entry fields where users can enter amounts

- Send on main screen 
- Receive / Custom
- Product / Edit product
- Product / Add product

Also changed product list on small devices to make it look less weird - only change is that the first element is 100px wide instead of 50 (which looked weird). 
And removed the leading 10px spacer which is not needed.

### 🙈 Screenshots

![Simulator Screen Shot - tester1 - 2021-01-22 at 13 59 34](https://user-images.githubusercontent.com/65412/105453359-f5a78100-5cba-11eb-969f-e9cba074ca76.png)

![Simulator Screen Shot - tester1 - 2021-01-22 at 13 59 39](https://user-images.githubusercontent.com/65412/105453370-f9d39e80-5cba-11eb-83d0-8cfb00fa7fef.png)
![Simulator Screen Shot - tester1 - 2021-01-22 at 13 59 47](https://user-images.githubusercontent.com/65412/105453372-fa6c3500-5cba-11eb-83dc-b5533b59b7b9.png)
![Simulator Screen Shot - tester1 - 2021-01-22 at 13 59 56](https://user-images.githubusercontent.com/65412/105453374-fb04cb80-5cba-11eb-9042-34132aa8b36d.png)
![Simulator Screen Shot - tester1 - 2021-01-22 at 14 00 00](https://user-images.githubusercontent.com/65412/105453376-fc35f880-5cba-11eb-9db2-1e93e32b36b4.png)
![Simulator Screen Shot - tester1 - 2021-01-22 at 14 00 17](https://user-images.githubusercontent.com/65412/105453378-fcce8f00-5cba-11eb-82da-8603d36d49db.png)
![Simulator Screen Shot - tester1 - 2021-01-22 at 14 00 23](https://user-images.githubusercontent.com/65412/105453380-fd672580-5cba-11eb-832b-5fd1eec0b000.png)
![Simulator Screen Shot - tester1 - 2021-01-22 at 14 00 41](https://user-images.githubusercontent.com/65412/105453382-fdffbc00-5cba-11eb-8f90-801b9f5e99bb.png)

### 👯‍♀️ Paired with

solo

n13 committed:
transfer screen integrates available balance under amount

- bugfixes and error entry handling
- layout for the under the amount elements

n13 committed:
better looks for small screens

n13 committed:
cleanup remove dependency on rate notifier

this becomes a more dumb widget which gets the rate from outside and as text

n13 committed:
build number

n13 committed:
Merge branch 'release/v1.4' into feature/transfer_screen

n13 committed:
more sensible quantity formatting with 2 trailing zeroes

honestly nobody cares about the last 2 digits - not in a normal case anyway.

n13 committed:
xcode stuff

n13 committed:
remove printout

n13 committed:
add progress bar to avoid jarring transition

n13 committed:
remove unused import

n13 committed:
version

n13 committed:
progress indicator since it takes a long time to generate a QR code

gguij004 reviewed (commented):
On this screenshot provided on this Pr, item 2 has USD and Item 1 SEEDS, but Item 29 is empty, any idea why this happen? 

![image](https://user-images.githubusercontent.com/51983923/105563308-9f066b00-5ceb-11eb-9ca4-7db580a347e9.png)

gguij004 reviewed (changes requested):
Validation was added so that a user wont be able to create a product with 0 price, yet I was able to create one by not adding any number on price. 

![image](https://user-images.githubusercontent.com/51983923/105565798-b008a980-5cf6-11eb-8d4b-5b01046a54e6.png)

gguij004 reviewed (changes requested):
You fixed this before, not sure why is broken now

![image](https://user-images.githubusercontent.com/51983923/105567003-33c59480-5cfd-11eb-816f-bb99656c632e.png)

n13 committed:
layout fix, size fix

small devices are only < 320 pixels
using Expanded expresses much more what we want - wrapped text that fills the space.

n13 committed:
fix validation bug

gguij004 reviewed (approved):
Receipt looks nicely even, for all devices tested (Iphone 12), Android (pixel 4, 4 WVGA - nexus)

n13 committed:
no validation on changed - the field validates on input but this gets triggered when the currency changes

n13 committed:
consistent number formatting, unit tests for number format
